### ISSUE TITLE
"invalid pixel format attribute" on OS X

### ISSUE THREAD

**Issue by dylanede** [tomaka/glutin#362]
All of the examples fail to run correctly on my OS X 10.8.5 machine at the moment, logging "invalid pixel format attribute" then failing to the create the OpenGL context and quitting. This may not be a recent change that caused this, since I have not touched glutin or glium since December.

**Comment by dylanede** [comment]
After upgrading to OS X 10.10, the problem no longer occurs. So this looks to be specific to 10.8.3 at least. Not sure whether you want to continue supporting that version.

**Comment by fkaa** [comment]
The only pixel format attributes that were added since December, 2014 seem to be `NSOpenGLPFAOpenGLProfile` (10.7). I don't have access to an earlier version of OS X so testing this seems tricky.

@dylanede did it panic and say `invalid pixel format attribute`? Replacing an enum with some random numbers only gives me this:
```rust
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: OsError("Couldn\'t create OpenGL context")', /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libcore/result.rs:768
Process didn't exit successfully: `target/debug/examples/vsync` (exit code: 101)
```

**Comment by dylanede** [comment]
@fkaa The message I received was very similar to that, only the previous the line of output would say

    invalid pixel format attribute

Just to be clear, the last time that I used Glutin on OS X 10.8.5 before encountering this problem was in December 2014, but I think the actual date was near the start of December, so if you're looking for relevant changes make sure to include those in December.

**Comment by fkaa** [comment]
Possible perpetrator: `NSOpenGLProfileVersion4_1Core`.

OS X 10.8.5 only supported up to 3.2 Core (`NSOpenGLProfileVersion3_2Core`) and `NSOpenGLProfileVersion4_1Core` is defined separately in rust-cocoa as `0x4100`. This could be a simple case of requesting a context which is not available (`NSOpenGLProfileVersion4_1Core` is default, afaik).

Possible solution: Check against current OS version and cap version accordingly (eg. on 10.8.5, `GlRequest::Latest` would request `NSOpenGLProfileVersion3_2Core` instead)

**Comment by fkaa** [comment]
Should be fixed in #470.
